src/native/minipal/memory.c

```diff
@@ -0,0 +1,40 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#include "memory.h"
+
+#ifdef HOST_WINDOWS
+#include <Windows.h>
+
+void* minipal_co_task_mem_alloc(size_t cb)
+{
+    return CoTaskMemAlloc(cb);
+}
+
+void minipal_co_task_mem_free(void* pv)
+{
+    CoTaskMemFree(pv);
+}
+#else
+// CoTaskMemAlloc always aligns on an 8-byte boundary.
+#define ALIGN 8
+
+void* minipal_co_task_mem_alloc(size_t cb)
+{
+    // Ensure malloc always allocates.
+    if (cb == 0)
+        cb = ALIGN;
+
+    // Align the allocation size.
```

Comment:
Can this just forward to malloc, just like the `Marshal.AllocCoTaskMem`?

The Windows PAL tried to be bug-for-bug compatible with real Windows in many places. I do not think we need to maintain that.